stl/inc/random

```diff
@@ -1349,7 +1349,10 @@ public:
 template <class _Engine, int _Px, int _Rx>
 class discard_block { // discard_block compound engine
 public:
-    using base_type   = _Engine;
+#if _HAS_TR1_NAMESPACE
+    using base_type _DEPRECATE_TR1_NAMESPACE = _Engine; // TR1-only typedef

```

Comment:
`discard_block` is entirely TR1-era machinery. Should we remove `discard_block_engine`'s dependence on it and wrap the entire class in `_HAS_TR1_NAMESPACE`? (No change requested.)

Reply:
Unfortunately, we need to keep `discard_block_engine`'s data member the wrong size (`int`) to preserve ABI, except when the class would have been utterly broken, which is why #4066 uses `_Discard_block_engine` as the base class conditionally.

We could remove the inheritance and switch the data member's type directly, to preserve the layout while altering the inheritance, but we haven't (yet) done that for `uniform_int` and `uniform_int_distribution` etc. I suppose if vNext is delayed enough, excising more TR1 over time would be worth it.

Reply:
Yeah, this would be nice if we ever have time, but certainly is more unrelated work than would be reasonable in this PR. I don't think we need a tracking issue either since we're well aware that we need to remove everything TR1-related.